packages/rocketchat-api/server/v1/groups.js

```diff
@@ -289,7 +289,12 @@ RocketChat.API.v1.addRoute('groups.list', { authRequired: true }, {
 	get() {
 		const { offset, count } = this.getPaginationItems();
 		const { sort, fields } = this.parseJsonQuery();
-		let rooms = _.pluck(RocketChat.models.Subscriptions.findByTypeAndUserId('p', this.userId).fetch(), '_room');
+		let rooms;
+		if (RocketChat.authz.hasPermission(this.userId, 'view-room-administration')) {
+			rooms = _.pluck(RocketChat.models.Subscriptions.findByType('p').fetch(), '_room');
```

Comment:
This will have duplicate rooms since the subscriptions are for each user and each room the user is part of. Use the `Rooms` model and find by the private type.